Installation/CMakeLists.txt

```diff
@@ -592,10 +592,8 @@ set(CGAL_INSTALL_LIB_DIR
 )
 
 if(CGAL_WIN32_CMAKE_ON_CYGWIN)
-  exec_program(
```

Comment:
@lrineau do you see any reason of the failure
```
  -- Installing: C:/dev/cygwin64/tmp/N1JWv/non_standard_install_qt
  /include/CGAL/Qt
  CMake Error at cmake_install.cmake:37 (file):
    file INSTALL cannot make directory
    "C:/dev/cygwin64/tmp/N1JWv/non_standard_install_qt
  
    /include/CGAL/Qt": Invalid argument.
```
 [here](https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-6.0-Ic-144/Installation/TestReport_Christo_MSVC-2022-Community-Release.gz)

I tried the following with cmake -P in the cygwin terminal
```
SET(CMAKE_INSTALL_PREFIX "/cygdrive/c/tmp")

exec_program(cygpath ARGS -w "${CMAKE_INSTALL_PREFIX}" OUTPUT_VARIABLE VAR1)


execute_process(COMMAND cygpath -w ${CMAKE_INSTALL_PREFIX} OUTPUT_VARIABLE VAR2)


message("VAR1: ${VAR1}")
message("VAR2: ${VAR2}")
```
 and got 
```
VAR1: C:\tmp
VAR2: C:\tmp
```

Reply:
I am not sure. Note that `file(TO_CMAKE_PATH ...)` is also involved, and that is important. Let file `test1.cmake` be:

```cmake
SET(CMAKE_INSTALL_PREFIX "/cygdrive/c/tmp")

exec_program(cygpath ARGS -w "${CMAKE_INSTALL_PREFIX}" OUTPUT_VARIABLE VAR1)


execute_process(COMMAND cygpath -w ${CMAKE_INSTALL_PREFIX} OUTPUT_VARIABLE VAR2)


message("VAR1: ${VAR1}")
message("VAR2: ${VAR2}")
```

and `test2.cmake` be:

```cmake
SET(CMAKE_INSTALL_PREFIX "/cygdrive/c/tmp")

exec_program(cygpath ARGS -w "${CMAKE_INSTALL_PREFIX}" OUTPUT_VARIABLE VAR1B)
execute_process(COMMAND cygpath -w ${CMAKE_INSTALL_PREFIX} OUTPUT_VARIABLE VAR2B)
file(TO_CMAKE_PATH ${VAR1B} VAR1)
file(TO_CMAKE_PATH ${VAR2B} VAR2)

message("VAR1: ${VAR1}")
message("VAR2: ${VAR2}")
```

then with `/usr/bin/cmake` from Cygwin:

```shell-session
$ cmake -P test1.cmake
VAR1: C:\tmp
VAR2: C:\tmp


Laurent Rineau@christo ~/tmp
$ cmake -P test2.cmake
VAR1: C;/tmp
VAR2: C;/tmp

```

but with a native CMake version:

```shell-session
Laurent Rineau@christo ~/tmp
$ /cygdrive/c/dev/CMake/bin/cmake.exe -P test1.cmake
VAR1: C:\tmp
VAR2: C:\tmp


Laurent Rineau@christo ~/tmp
$ /cygdrive/c/dev/CMake/bin/cmake.exe -P test2.cmake                                                                                                                                                                
VAR1: C:/tmp
VAR2: C:/tmp


```

The difference is with `test2.cmake` that the `:` is considered to be a separator (like in `$PATH`) by `TO_CMAKE_PATH`, and that converts the argument to a list (with `;` as separator).

With the native CMake, `C:` is recognized.

But the testsuite uses native CMake.

And anyway that does not explains why the path

```
C:/dev/cygwin64/tmp/N1JWv/non_standard_install_qt
/include/CGAL/Qt
```

was created (probably with a `\n` in the middle).

Reply:
A more interesting test is:

```cmake
SET(CMAKE_INSTALL_PREFIX "/cygdrive/c/tmp")

exec_program(cygpath ARGS -w "${CMAKE_INSTALL_PREFIX}" OUTPUT_VARIABLE VAR1B)     
execute_process(COMMAND cygpath -w ${CMAKE_INSTALL_PREFIX} OUTPUT_VARIABLE VAR2B) 
file(TO_CMAKE_PATH ${VAR1B} VAR1)
file(TO_CMAKE_PATH ${VAR2B} VAR2)

message("VAR1: ${VAR1}")
message("VAR2: ${VAR2}")
message("END")
```

because it shows the `\n` in the variable created by `execute_process`:

```shell-session
Laurent Rineau@christo ~/tmp
$ /cygdrive/c/dev/CMake/bin/cmake.exe -P test2.cmake
VAR1: C:/tmp
VAR2: C:/tmp

END

```

Reply:
So the solution is: add `OUTPUT_STRIP_TRAILING_WHITESPACE` as argument of `execute_process`.